### PR TITLE
feat: add caching for owm segment

### PR DESCRIPTION
**Pull request by lnu**
default timeout of 10 minutes
Original concept by boarder2

### Prerequisites

- [x] I have read and understood the `CONTRIBUTING` guide
- [x] The commit message follows the [conventional commits][cc] guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes/features)

### Description

Original pull by @boarder2 (#946). Relies on the new global cache by @JanDeDobbeleer.

[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

File: src/segment_owm.go [hunk 1]
```diff
@@ -61,22 +63,57 @@ func (d *owm) string() string {
...
 	timeout := d.props.getInt(HTTPTimeout, DefaultHTTPTimeout)
+	cachetimeout := int64(d.props.getInt(CACHETIMEOUT, 10))
+
+	d.url = fmt.Sprintf("http://api.openweathermap.org/data/2.5/weather?q=%s&units=%s&appid=%s", location, units, apikey)
+
+	if cachetimeout > 0 {
+		// check if data stored in cache
+		val, found := d.env.cache().get("owm_response")
+		// we got something from te cache
+		if found {
```
Comment (JanDeDobbeleer) [inline src/segment_owm.go:81]: Do an early return with `!found`

File: src/segment_owm.go [hunk 2]
```diff
@@ -21,6 +21,8 @@ const (
 	LOCATION Property = "location"
 	// UNITS openweathermap units
 	UNITS Property = "units"
+	// CACHETIMEOUT cache timeout
+	CACHETIMEOUT Property = "cachetimeout"
```
Comment (JanDeDobbeleer) [inline src/segment_owm.go]: Mismatch in doc and value `cache_timeout`, would also use the underscore here.

File: src/segment_owm.go [hunk 3]
```diff
@@ -61,22 +63,57 @@ func (d *owm) string() string {
...
-		return err
+		return new(OWMDataResponse), err
 	}
 	q := new(OWMDataResponse)
 	err = json.Unmarshal(body, &q)
+	if err != nil {
+		return new(OWMDataResponse), err
+	}
+
+	if cachetimeout > 0 {
```
Comment (JanDeDobbeleer) [inline src/segment_owm.go]: Also use an early return here

**Review by JanDeDobbeleer — CHANGES_REQUESTED** [review]

File: src/segment_owm.go (hunk 2 above)
Comment (lnu) [reply src/segment_owm.go]: my bad... and the schema.json property has also been added on the wrong segment.

**Review by JanDeDobbeleer — APPROVED** [review]

File: src/segment_owm.go (hunk 1 above)
Comment (lnu) [reply src/segment_owm.go:81]: in that case, if not found in the cache we continue the flow and call the service. If found we exit. Does an early return with !found apply here?

**Comment by boarder2** [comment]
Thanks for taking this and running with it @lnu ! Sorry I wasn't able to get back to it quickly!

**Pull request merged**